src/log-store/src/kafka/client_manager.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::time::Duration;
+
+use common_config::wal::{KafkaConfig, KafkaWalTopic as Topic};
+use dashmap::mapref::entry::Entry as DashMapEntry;
+use dashmap::DashMap;
+use rskafka::client::partition::{PartitionClient, UnknownTopicHandling};
+use rskafka::client::producer::aggregator::RecordAggregator;
+use rskafka::client::producer::{BatchProducer, BatchProducerBuilder};
+use rskafka::client::{Client as RsKafkaClient, ClientBuilder};
+use rskafka::BackoffConfig;
+use snafu::ResultExt;
+
+use crate::error::{BuildClientSnafu, BuildPartitionClientSnafu, Result};
+
+// Each topic only has one partition for now.
+// The `DEFAULT_PARTITION` refers to the index of the partition.
+const DEFAULT_PARTITION: i32 = 0;
+
+/// Arc wrapper of Client.
+pub(crate) type ClientRef = Arc<Client>;
```

Comment:
Looks like the `ClientRef` is unnecessary. The client all fields are wrapped with `Arc`, so it is cloneable.

Reply:
Let the `Client` derive `Clone` and removed the `ClientRef`.